mne/preprocessing/tests/test_ica.py

```diff
@@ -143,12 +143,46 @@ def test_ica_simple(method):
     data = np.dot(A, S)
     ica = ICA(n_components=n_components, method=method, random_state=0)
     ica._fit(data, n_components, 0)
-    transform = np.dot(np.dot(ica.unmixing_matrix_, ica.pca_components_), A)
+    transform = ica.unmixing_matrix_ @ ica.pca_components_ @ A
     amari_distance = np.mean(np.sum(np.abs(transform), axis=1) /
                              np.max(np.abs(transform), axis=1) - 1.)
     assert amari_distance < 0.1
 
 
+@requires_sklearn
+@pytest.mark.parametrize('n_pca_components', (8, 9, 10))
+@pytest.mark.parametrize('max_pca_components', [
+    9,
+    pytest.param(0.9999, marks=pytest.mark.xfail(reason='XXX should work')),
```

Comment:
@hoechenberger this should pass for `n_pca_components=8` and `9` (and eventually probably error for `10` but it fails currently. This suggests that the `max_pca_components=0.9999` component picking does not work right.